hudi-common/src/main/java/org/apache/hudi/common/schema/HoodieSchemaField.java

```diff
@@ -0,0 +1,392 @@
+    return of(name, schema, null, null);
+  }
+
+  /**
+   * Creates a new HoodieSchemaField with the specified name, schema, and doc.
+   *
+   * @param name   the name of the field
+   * @param schema the schema of the field
+   * @param doc    the documentation string
+   * @return new HoodieSchemaField instance
+   */
+  public static HoodieSchemaField of(String name, HoodieSchema schema, String doc) {
+    return of(name, schema, doc, null);
+  }
+
+  /**
+   * Creates a metadata field for Hudi internal use.
+   * This is a convenience method for creating fields that are part of Hudi's metadata.
+   *
+   * @param name   the metadata field name
+   * @param schema the metadata field schema
+   * @return new HoodieSchemaField configured as a metadata field
+   * @throws IllegalArgumentException if name is null/empty or schema is null
+   */
+  public static HoodieSchemaField createMetadataField(String name, HoodieSchema schema) {
+    ValidationUtils.checkArgument(name != null && !name.isEmpty(), "Metadata field name cannot be null or empty");
+    ValidationUtils.checkArgument(schema != null, "Metadata field schema cannot be null");
+
+    return HoodieSchemaField.of(name, schema, "Hudi metadata field: " + name, HoodieJsonProperties.NULL_VALUE);
+  }
+
+  /**
+   * Returns the name of this field.
+   *
+   * @return the field name
+   */
+  public String name() {
+    return avroField.name();
+  }
+
+  /**
+   * Returns the schema of this field.
+   *
+   * @return the field schema as HoodieSchema
+   */
+  public HoodieSchema schema() {
+    return hoodieSchema;
+  }
+
+  /**
+   * Returns the documentation string for this field.
+   *
+   * @return Option containing the documentation string, or Option.empty() if none
+   */
+  public Option<String> doc() {
+    return Option.ofNullable(avroField.doc());
+  }
+
+  /**
+   * Returns the default value for this field.
+   *
+   * @return Option containing the default value, or Option.empty() if none
+   */
+  public Option<Object> defaultVal() {
+    if (avroField != null && avroField.hasDefaultValue()) {
+      return Option.of(avroField.defaultVal());
+    }
+    return Option.empty();
+  }
+
+  /**
+   * Returns the sort order for this field.
+   *
+   * @return the field order
+   */
+  public HoodieFieldOrder order() {
+    return HoodieFieldOrder.fromAvroOrder(avroField.order());
+  }
+
+  /**
+   * Returns the position of this field within its enclosing record.
+   *
+   * @return the field position (0-based index)
+   */
+  public int pos() {
+    return avroField.pos();
+  }
+
+  /**
+   * Checks if this field has a default value.
+   *
+   * @return true if the field has a default value
+   */
+  public boolean hasDefaultValue() {
+    return avroField.hasDefaultValue();
+  }
+
+  /**
+   * Returns custom properties attached to this field.
+   *
+   * @return map of custom properties
+   */
+  public java.util.Map<String, Object> getObjectProps() {
```

Comment:
Can this be imported instead of using the FQDN?

Reply:
sure